pysand/erosion.py

```diff
@@ -1,7 +1,32 @@
 import numpy as np
+import pandas as pd
+import logging
+import pysand.exceptions as exc
 
+logger = logging.getLogger(__name__)
 # Models from DNVGL RP-O501, equation references in parenthesis
 
+
+def validate_inputs(v_m, Q_s, rho_m=30, mu_m=1.5e-5, GF=3, D=0.5, d_p=1):
+    """
+    Validation of all input parameters that go into erosion models;
+    Some parameters are passed as Optional, since models take only specific input arguments.
+    """
+
+    for i in [v_m, Q_s, rho_m, mu_m]:
+        if not isinstance(i, (float, int, np.ndarray, pd.Series)):
+            raise exc.FunctionInputFail('{} is not a number or pandas series'.format(i))
+
+    for j in [GF, D, d_p]:
+        if not isinstance(j, (int, float)):
+            raise exc.FunctionInputFail('{} is not a number'.format(j))
+
+    if (d_p < 0.02) or (d_p > 5):
+        logger.warning('The particle diameter, d_p, is outside RP-O501 model boundaries.')
+    if (D < 0.01) or (D > 1):
+        logger.warning('The pipe inner diameter, D, is outside RP-O501 model boundaries.')
```

Comment:
logger.warning('The pipe inner diameter (D = %s m) is outside RP-O501 model boundaries (0.01 - 1 m).', d_p)